src/test/mathutiltest.cpp

```diff
@@ -89,5 +89,17 @@ TEST_F(MathUtilTest, Denormal) {
 #endif
 }
 
+TEST_F(MathUtilTest, DoubleValues) {
+    // This verifies that the infinity value can be copied into -ffastmath code
+    if (std::numeric_limits<double>::is_iec559) {
+        union {
+            long long int_value;
+            double double_value;
+        } inf_union;
+        inf_union.double_value = util_double_infinity();
+        // IEC 559 (IEEE 754) Infinity
+        EXPECT_EQ(inf_union.int_value, 0x7FF0000000000000);
+    }
+}
```

Comment:
`is_iec559` should be a static assert IMO. otherwise the test just passes when the double is not a proper float. Moreover type punning using a union like this is also highly undefined behavior! You'll need `std::bit_cast` of `std::memcpy`, those are the only ways to do type punning safely.